### DISCUSSION TITLE
Acces token to get zip file

### DISCUSSION THREAD

**Discussion by brunobart** [dcm4che/dcm4chee-arc-light#4464] [Q&A] [answered]
Hi,

I use DCM4CHE with a symfony application that I coded. One of my functions is to recover study files in zip form. I coded it in 5.14 but I have some difficulties with version 5.31.

***1 - recovery of a token***
I write this documentation https://github.com/dcm4che/dcm4chee-arc-light/wiki/Getting-OIDC-Access-Token-using-curl to how get a valid token with an Authenticator "client id and secret". 
It work but i didn't know how to use my token to recover study files in zip file.

***2 - Retrieving the zip of a study by web service***
The url used in version 5.14 is not still valid or i forget something in the keycloak client parameters.
This url
https://{{MyIp}}:8443/dcm4chee-arc/aets/DCM4CHEE/rs/studies/{{ studyIuid}}?accept=application/zip;transfer-syntax=*&access_token={{ token}}
resent "Unauthorized"
Can you help me ? Is it the url ou the keycloak configuration witch is wrong ?

I thank in advance for the answers
Best regard

**Comment by vrindanayak** [comment, marked as answer]
You don't need to write a new page again for an existing wiki - see https://github.com/dcm4che/dcm4chee-arc-light/wiki/Get-OIDC-Access-Token-using-curl#request-an-access-token-using-oauth-20-client-credentials-grant

As for WADO-RS service you want to use, refer https://petstore.swagger.io/index.html?url=https://dcm4che.github.io/dcm4chee-arc-light/swagger/openapi.json#/WADO-RS/RetrieveStudy

**Note : access token should be passed in HTTP Request header, not as a query param!**
```
curl -k -H "Authorization: Bearer {access-token}" -X GET 'https://{MyIp}:8443/dcm4chee-arc/aets/DCM4CHEE/rs/studies/{studyIuid}?accept=application/zip;transfer-syntax=*'
```
